app/api/ada/lib/storage/adaMigration.js

```diff
@@ -35,39 +41,63 @@ export async function migrateToLatest(localStorageApi: LocalStorageApi, isImport
     * Therefore mobx may not notice the change as expected
     */
 
-  const migrationMap: { [ver: string]: (() => Promise<void>) } = {
-    '=0.0.1': async () => await testMigration(localStorageApi),
-    '<1.9.0': async () => await moveStorage(localStorageApi, isImported),
-    '<1.4.0': bip44Migration
-  };
+  const migrationMap: Array<[string, () => Promise<boolean>]> = [
+    ['=0.0.1', async () => await testMigration(localStorageApi)],
+    /**
+     * We changed the place where Yoroi data is stored so  we must process this migration first
+     * to ensure other migrations look at the right place for storage
+     */
+    ['<1.9.0', async () => await moveStorage(localStorageApi)],
+    ['<1.4.0', bip44Migration],
+  ];
 
-  for (const key of Object.keys(migrationMap)) {
-    if (semver.satisfies(lastLaunchVersion, key)) {
-      Logger.info(`Migration started for ${key}`);
-      await migrationMap[key]();
+  let appliedMigration = false;
+  for (const entry of migrationMap) {
+    if (satisfies(lastLaunchVersion, entry[0])) {
+      Logger.info(`Migration started for ${entry[0]}`);
+      const applied = await entry[1]();
+      if (applied) {
+        Logger.info(`Applied migration for ${entry[0]}`);
+      }
+      appliedMigration = appliedMigration || applied;
     }
   }
+
+  return appliedMigration;
 }
 
 /**
  * We use this as a dummy migration so that our tests can verify migration is working correctly
  */
-async function testMigration(localStorageApi: LocalStorageApi): Promise<void> {
+async function testMigration(localStorageApi: LocalStorageApi): Promise<boolean> {
   // changing the locale is something we can easily detect from our tests
   Logger.info(`Starting testMigration`);
   // Note: mobx will not notice this change until you refresh
   await localStorageApi.setUserLocale('ja-JP');
+  return true;
 }
 
 /**
  * Previous version of Yoroi used window.localStorage
  * since version 1.9 this storage needs to be moved to storage.local.
  * This function must go before other modifications because they will work over storage.local.
  */
-async function moveStorage(localStorageApi: LocalStorageApi, isImported: boolean): Promise<void> {
-  if (isImported) return;
+async function moveStorage(localStorageApi: LocalStorageApi): Promise<boolean> {
+  // Note: this function assumes nobody is using Yoroi as a  website (at least before 1.9.0)
   const oldStorage = await localStorageApi.getOldStorage();
+  if (oldStorage.length === 0) {
+    return false;
+  }
+  if (oldStorage.length === 1 && oldStorage[OPEN_TAB_ID_KEY] != null) {
+    // old storage may contain just the tab id
+    // since this field is re-generated every time the app launches in the right storage
+    // we can just clear it
+    oldStorage.clear();
```

Comment:
I didn’t get why we should clean it up.

Reply:
No particular reason. If you don't clean it up you will have an unused `OPEN_TAB_ID_KEY` in your `localStorage` forever but it's not really a  big deal either